app/models/nic/base.rb

```diff
@@ -33,6 +33,8 @@ class Base < ActiveRecord::Base
     validates :domain_id, :presence => true, :if => Proc.new { |nic| nic.host && nic.host.managed? && nic.primary? }
     validates :ip, :presence => true, :if => Proc.new { |nic| nic.host && nic.host.managed? && nic.require_ip_validation? }
 
+    validate :validate_host_taxonomy
```

Comment:
Fair enough, that sounds like a very unusual case, so great catch :smile: 

Can you leave a comment at the `validate_host_taxonomy` method explaining that it's that way to ensure DB consistency when turning on/off taxonomies?